ISS/models/core_models.py

```diff
@@ -221,13 +222,27 @@ def can_auto_anonymize(self):
     @transaction.atomic
     def merge_into(self, other):
         """
-        Reassigns all the records that tie to this user to another one and
+        Reassigns all (<-- lies) the records that tie to this user to another one and
         disables this user afterwards.
         """
         
         Thread.objects.filter(author=self).update(author=other)
+
+        Post.objects.filter(author=self).update(posted_from='127.0.0.1')
         Post.objects.filter(author=self).update(author=other)
 
+        PostSnapshot.objects.filter(obsolesced_by=self).update(obsolescing_ip='127.0.0.1')
+        PostSnapshot.objects.filter(obsolesced_by=self).update(obsolesced_by=other)
+
+        for post in Post.objects.all():
+            post.content = post.content.replace("author=\"" + self.username + "\"", "author=\"" + str(other) + "\"")
+            post.save()
+
+        PrivateMessage.objects.filter(receiver=self).delete()
+        PrivateMessage.objects.filter(sender=self).delete()
+
+        PollVote.objects.filter(voter=self).update(voter=other)
```

Comment:
We should probably just delete votes. This reassignment to the target account makes it possible we'll end up with two votes from the target account on the same poll which I believe will end up being a unique constraint violation and will just fail the whole transaction (meaning the merge will get reversed after holding a bunch of DB locks for a long time)

Reply:
Delete pushed